Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/NameAndParametersListView.cs

```diff
@@ -41,20 +41,25 @@ public void OnAddElement(string name)
             m_ListProperty.serializedObject.ApplyModifiedProperties();
         }
 
-        private void MoveElement(int index, bool up)
+        private void MoveElementUp(int index)
         {
             var newIndex = index - 1;
-            if (!up)
-                newIndex = index + 1;
+            SwapElement(index, newIndex);
+        }
+
+        private void MoveElementDown(int index)
+        {
+            var newIndex = index + 1;
             SwapElement(index, newIndex);
         }
 
         private void SwapElement(int oldIndex, int newIndex)
         {
-            var interactionsOrProcessorsList = NameAndParameters.ParseMultiple(m_ListProperty.stringValue).ToArray();
-            newIndex = Math.Clamp(newIndex, 0, interactionsOrProcessorsList.Length - 1);
-            MemoryHelpers.Swap(ref interactionsOrProcessorsList[oldIndex], ref interactionsOrProcessorsList[newIndex]);
-            m_ListProperty.stringValue = ToSerializableString(interactionsOrProcessorsList);
+            var interactionsOrProcessors = NameAndParameters.ParseMultiple(m_ListProperty.stringValue).ToArray();
+            if (interactionsOrProcessors.Length == 0 || newIndex < 0 || newIndex >= interactionsOrProcessors.Length)
```

Comment:
Looks good. I think we should probably check oldIndex too
If we never expect this to be out of range we should probably assert so they get flagged.